src/test/java/com/github/btror/mcpathfinding/simulation/tests/SimulationTest.java

```diff
@@ -0,0 +1,114 @@
+        for (int i = 1; i < simulation.getSimulationPath().size() - 1; i++) {
+            Integer[] path = simulation.getSimulationPath().get(i);
+            assertEquals(simulation.getNodeSnapshot()[path[0]][path[1]][path[2]].getType(), 3);
+        }
+    }
+
+    @Test
+    public void testBeamSearch() {
+        Simulation simulation = new BeamSearch();
+        simulation.setBeamWidth(5);
+
+        mockSimulationSnapshot = createSimulationSnapshot();
+        simulation.setSimulationSnapshot(mockSimulationSnapshot);
+
+        int[] simulationStart = {0, 0, 0};
+        simulation.setSimulationStrikeStart(simulationStart);
+
+        int[] simulationTarget = {5, 0, 0};
+        simulation.setSimulationStrikeTarget(simulationTarget);
+
+        simulation.start();
+        simulation.findPath();
+
+        assertEquals(simulation.getNodeSnapshot()[0][0][0].getType(), 4);
+        assertEquals(simulation.getNodeSnapshot()[5][0][0].getType(), 5);
+        for (int i = 1; i < simulation.getSimulationPath().size() - 1; i++) {
+            Integer[] path = simulation.getSimulationPath().get(i);
+            assertEquals(simulation.getNodeSnapshot()[path[0]][path[1]][path[2]].getType(), 3);
+        }
+    }
+
+    @Test
+    public void testGreedyBestFirstSearch() {
+        Simulation simulation = new GreedyBestFirstSearch();
+
+        mockSimulationSnapshot = createSimulationSnapshot();
+        simulation.setSimulationSnapshot(mockSimulationSnapshot);
+
+        int[] simulationStart = {0, 0, 0};
+        simulation.setSimulationStrikeStart(simulationStart);
+
+        int[] simulationTarget = {5, 0, 0};
+        simulation.setSimulationStrikeTarget(simulationTarget);
+
+        simulation.start();
+        simulation.findPath();
+
+        assertEquals(simulation.getNodeSnapshot()[0][0][0].getType(), 4);
+        assertEquals(simulation.getNodeSnapshot()[5][0][0].getType(), 5);
+        for (int i = 1; i < simulation.getSimulationPath().size() - 1; i++) {
+            Integer[] path = simulation.getSimulationPath().get(i);
+            assertEquals(simulation.getNodeSnapshot()[path[0]][path[1]][path[2]].getType(), 3);
+        }
+    }
+
+    /**
+     * Simulation Integer Mappings:
+     * 0 = open space (can be explored in pathfinding)
+     * 1 = blocked space (cannot be explored in pathfinding)
+     * 2 = visited space (already explored in pathfinding)
+     * 3 = final path space (space in the final path)
+     * 4 = start space
+     * 5 = target space
+     */
+    public int[][][] createSimulationSnapshot() {
+        int[][][] simulationSnapshot = new int[10][10][10];
+        for (int i = 0; i < simulationSnapshot.length; i++) {
+            for (int j = 0; j < simulationSnapshot[0].length; j++) {
+                for (int k = 0; k < simulationSnapshot[0][0].length; k++) {
+                    simulationSnapshot[i][j][k] = 0;
+                }
+            }
+        }
+
+        return simulationSnapshot;
+    }
```

Comment:
make some of the nodes blockers to really see the difference between the algorithms